operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/InformerEventSource.java

```diff
@@ -0,0 +1,85 @@
+package io.javaoperatorsdk.operator.processing.event.internal;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedInformer;
+import io.fabric8.kubernetes.client.informers.cache.Store;
+import io.javaoperatorsdk.operator.processing.event.AbstractEventSource;
+
+public class InformerEventSource<T extends HasMetadata> extends AbstractEventSource {
```

Comment:
Wonder if we should have a simple constructor that hides the details about creating an informer

Reply:
Not sure if that is desirable, fabric8 now supports shared informers, so it might be also common case that those are shared between multiple EventSource. 
Also I don't see how that would look like, based on the current fabric8 api for informers.

Reply:
Correction: Wonder if we should **ALSO** have a simple constructor that hides the details about creating an informer, something like:

```java
InformerEventSource(KubernetesClient client, Class<T> type) {
    this.informer = client.informers().sharedIndexInformerFor(type.class, 0);
}
```

Reply:
Added additional constructors.